docs/index.rst

```diff
@@ -3,17 +3,95 @@
 
 The ``pytest`` plugin system
 ****************************
+
 ``pluggy`` is the crystallized core of `plugin management and hook
-calling`_ for `pytest`_.
+calling`_ for `pytest`_. It gives users the possibility to extend or modify
+the behaviour of a host program (e.g. `pytest`_, `tox`_ or `devpi`_) by
+installing a `plugin` for that program. The plugin code will run as part of
+normal program execution changing or enhancing certain aspects of it.
 
 In fact, ``pytest`` is itself composed as a set of ``pluggy`` plugins
 which are invoked in sequence according to a well defined set of protocols.
 Some `200+ plugins`_ use ``pluggy`` to extend and customize ``pytest``'s default behaviour.
 
 In essence, ``pluggy`` enables function `hooking`_ so you can build "pluggable" systems.
 
-How's it work?
---------------
+Introduction
+------------
+
+To explain how pluggy can be put to use we need to introduce three roles and explain their interaction with pluggy:
+
+* The `hoster` - the person who wants to add extensibility to their program
+* The `implementer` - the person who wants to extend the `host` functionality by writing a plugin
```

Comment:
The `implementer` - the plugin itself which extends or modifies the `host`s functionality and/or behaviour